### PR TITLE
Edits and #ifdefs to get SLiM and SLiMgui to build on Windows 10

### PR DESCRIPTION
**Pull request by rdinnager**
The rest of the Windows port changes.

### CODE REVIEW

**Review by bhaller — COMMENTED** [review]
Hey.  I just updated the windows branch with the various changes I mention in my review comments here.  You should be able to prune this PR down to a pretty small number of changes now.  Let me know if you have any questions.

**Comment by bhaller** [comment]
You'll want to rebase to the latest windows_port branch now, however you do that...

File: QtSLiM/QtSLiMAbout.cpp [hunk 1]
```diff
@@ -17,6 +17,9 @@
 //
 //	You should have received a copy of the GNU General Public License along with SLiM.  If not, see <http://www.gnu.org/licenses/>.
 
+#ifdef _WIN32
+#define GNULIB_NAMESPACE gnulib
```
[the remark this answers is not in the document]
Comment (rdinnager) [reply QtSLiM/QtSLiMAbout.cpp:21]: I think this is possible yes. I will try it.

File: core/slim_eidos_block.cpp [hunk 2]
```diff
@@ -23,6 +23,9 @@
 #include <algorithm>
 #include <vector>
 
+#ifdef _WIN32
+#define strtoq strtoll
```
[the remark this answers is not in the document]
Comment (rdinnager) [reply core/slim_eidos_block.cpp:27]: Yes, unfortunately `strtoq` is simply missing from MinGW.. who knows why. And unfortunately, `gnulib` does not have a fix for this particular one (https://www.gnu.org/software/gnulib/manual/html_node/strtoq.html). This change was suggested after some googling but yes, I have no idea whether it is really a good idea, or if there is a better solution. Your idea sounds good.

**Comment by rdinnager** [comment]
Okay, all of these edits should be pretty easy to get done. I'm about to head home, so this might not get done until tomorrow. Thanks for all your help so far, I've learned a lot about C++ in the last week or two!

**Comment by bhaller** [comment]
> Okay, all of these edits should be pretty easy to get done. I'm about to head home, so this might not get done until tomorrow. Thanks for all your help so far, I've learned a lot about C++ in the last week or two!

Sounds good!  Thanks for doing this!  I'll ponder the strtoq() problem and make a decision.

File: core/slim_eidos_block.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bhaller) [reply core/slim_eidos_block.cpp:27]: Apparently strtoq() has been deprecated in BSD.  strtoll() should be equivalent as long as the `long long` type is 64-bit and not 128-bit or something.  As far as I can tell from googling, `long long` is _allowed_ by the standard to be more than 64-bit, but in practice it never is, on any platform SLiM is likely to be used on.  So strtoll() should be an acceptable substitute for everybody; I'll switch over to it in my next commit.  Man, the built-in types in C are a mess.

**Comment by rdinnager** [comment]
Okay, I think I've made all the changes, but I am getting the same problem where changes to the base branch are not being updated in the diffs. Apparently this is a known issue (and github says it is intended behaviour: https://github.com/isaacs/github/issues/750). One workaround suggested in that issue is to change the base branch to a random branch, then change it back. As you can see I tried that but unfortunately it didn't change anything. I think the mistaken diffs should be harmless, that is, if you merge the pull request we should end up with what we expect. But it is annoying, so I will probably close this pull request again, and make a new one, which seems to be the only sure way to get rid of the issue.

**Comment by bhaller** [comment]
Ugh.  Yes, I'd recommend starting a new PR, and then I'll review it and hopefully can accept it before I need to do more commits on my end.  I don't trust git to do the right thing when merging in this situation; I have tried that before, and have ended up with a huge mess of conflicts and duplicated code all over the place.  Make backups, tread carefully.  Maybe @petrelharp has advice for how to rebase your PRs without getting into this mess, his git-fu is strong.

**Comment by rdinnager** [comment]
Sorry I don't currently use slack. I have an account but I use it so rarely it takes me half an hour to figure out how to use it each time I try. I could try again if it would be helpful.

Anyway, I will close this now and make a new pull request right away. I think it is ready for another review. I added some code to get the default working directory to work in SLiMgui on Windows, and some code to get the temp directory for a user in Windows.

**Comment by bhaller** [comment]
OK, sounds good.  No worries on Slack if you don't presently use it (but I recommend setting it up for your lab when you start your new position, it's a very useful tool for coordinating lab communication).  I'm going out for a bit to run errands, will review your PR when I get back (ish).  Sounds good.

**Pull request closed without merging**